docs/vNext/DeclarativeWorkloadModule.md

```diff
@@ -0,0 +1,190 @@
+- name: randomNumber
+  type: uniform_random
+  options:
+    min: 0
+    max: 100
+```
+##### Attributes
+
+|Property   |Type  |Description                                        |
+|:----------|:-----|:--------------------------------------------------|
+|type       |string|Assigned the value `uniform_random`                |
+|name       |string|Parameter Name                                     |
+|options.min|number|Mininum inclusive range for generated random number|
+|options.max|number|Maximum inclusive range for generated random number|
+
+#### Parameter Reference
+
+Value Provider format for referencing a `module.arguments` item.
+
+##### Example
+```yaml
+- name: referenceNumber
+  type: parameter_reference
+  options:
+    name: marbleNumber #assuming this is defined under round.arguments
+```
+##### Attributes
+
+|Property|Type|Description|
+|:-------|:---|:----------|
+|type    |string|Assigned the value `parameter_reference`|
+|name    |string|Parameter name|
+|options.name|string|Should be a variable name specified under `module.arguments`|
+
+#### Variable Reference
+
+Value Provider format for referencing a Caliper workload variable.
+
+##### Example
+```yaml
+- name: referenceNumber
+  type: variable_reference
+  options:
+    name: txIndex #a Caliper workload parameter
+```
+##### Attributes
+
+|Property|Type|Description|
+|:-------|:---|:----------|
+|type    |string|Assigned the value `variable_reference`.|
+|name    |string|Parameter name|
+|options.name    |string|Parameter Name. Should refer to a Caliper variable. |
+
+#### List Element
+
+Value provider format for selecting an item from a given list.
+##### Example
+
+```yaml
+- name: selectedColor
+  type: list_element
+  options:
+    list: ['red', blue', 'green'],
+    selector:
+      type: parameter_reference
+      options:
+        name: marblePrefix #assuming this is defined under module.arguments
+```
+##### Attributes
+
+|Property|Type|Description|
+|:-------|:---|:----------|
+|type    |string|Assigned the value `list_element`|
+|name    |string|Parameter Name|
+|options.list|list|List from which element is chosen.|
+|options.selector|object|Contains information about `variable_reference` and `parameter_reference` item used to select list item.|
+
+
+#### Formatted String
+
+Value provider format for generating formatted strings.
+
+##### Example
+
+```yaml
+- name: generatedString
+  type: formatted_string
+  options:
+    format: '{1}'
+    parts:
+      - type: parameter_reference
+        options:
+          name: marbleIndex
+```
+
+##### Attributes
+
+|Property|Type  |Description|
+|:-------------|:-----|:----------|
+|type          |string|Assigned the value `formatted_string`|
+|name          |string|Parameter Name|
+|options.format|string|Specifies format and placeholders for variables. Placeholders are specified using this syntax: `{variable_no}`.|
+|options.parts |list  |Specifies variable and parameter reference value providers for use in string|
```

Comment:
A section is still needed at the end to demonstrate how to create a derived class based on this workload module, and which function should be implemented.